src/web-components/img-slider/img-slider.js

```diff
@@ -27,18 +145,80 @@ class ImgSlider extends HTMLElement {
         this.attachShadow({ mode: 'open' });
         this.shadowRoot.appendChild(template.content.cloneNode(true));
 
-        const slides = this.querySelectorAll('[slot="slide"]');
-        const thumbnails = this.querySelectorAll('[slot="thumbnail"]');
+        this.slidesContainer = this.shadowRoot.querySelector('.slides');
+        this.thumbnailsContainer = this.shadowRoot.querySelector('.thumbnails');
 
-        slides.forEach((slide) => {
-            slide.classList.add('hidden');
-            slide.classList.add('slide');
-        })
+        this.slideSlot = this.shadowRoot.querySelector('slot[name="slide"]');
+        this.images = this.slideSlot.assignedElements();
+        this.thumbnails = this.shadowRoot.querySelector('slot[name="thumbnail"]').assignedElements();
+        this.thumbnails[0].classList.add('active');
+
+        this.previousBtn = this.shadowRoot.querySelector('.previous');
+        this.nextBtn = this.shadowRoot.querySelector('.next');
+
+        this.previousBtn.addEventListener('click', (e) => this.navigateImages(e));
+        this.nextBtn.addEventListener('click', (e) => this.navigateImages(e));
+        this.slidesContainer.addEventListener('scroll', (e) => this.thumbnailNavigation(e));
+        this.thumbnailsContainer.addEventListener('click', (e) => this.thumbnailNavigation(e));
+    }
+
+    navigateImages(e) {
+        const clickedBtn = e.target.closest('custom-button').classList[0];
+        const scrollPosition = this.slidesContainer.scrollLeft;
 
-        thumbnails.forEach((thumbnail) => {
-            thumbnail.classList.add('hidden');
+        if (clickedBtn === 'next') {
+            this.slidesContainer.scrollTo({ left: scrollPosition + 320, behavior: 'smooth' });
+        }
+
+        if (clickedBtn === 'previous') {
```

Comment:
Ook is het beter om niet naar een hardcoded stukje text (in dit geval class). Vind in dit geval netter om allebei de buttons apart te definieren.